### PR TITLE
Use the `empty_row` table function implicitly in statements without `FROM`

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                        }
+                    };
+                }
+            };
+        }
+
+        @Override
+        public TableInfo createTableInfo(ClusterService clusterService) {
+            final String localNodeId = clusterService.localNode().getId();
+            return new StaticTableInfo(TABLE_IDENT, new LinkedHashMap<>(0), null, Collections.emptyList()) {
```
Comment (seut) [inline sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java]: I think there is no need for an LinkedHashMap, wouldn't Collections.emptyMap() work?

File: sql/src/test/java/io/crate/operation/tablefunctions/AbstractTableFunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private SqlExpressions sqlExpressions;
+    private Functions functions;
+
+    @Before
+    public void prepareFunctions() throws Exception {
+        sqlExpressions = new SqlExpressions(ImmutableMap.of(QualifiedName.of("t"), mock(DocTableRelation.class)));
+        functions = sqlExpressions.getInstance(Functions.class);
+    }
+
+    public Bucket execute(String expr) {
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/tablefunctions/AbstractTableFunctionsTest.java]: can be protected

File: sql/src/test/java/io/crate/operation/tablefunctions/AbstractTableFunctionsTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        Symbol functionSymbol = sqlExpressions.asSymbol(expr);
+        functionSymbol = sqlExpressions.normalize(functionSymbol);
+        Function function = (Function) functionSymbol;
+        FunctionIdent ident = function.info().ident();
+        TableFunctionImplementation tableFunction = (TableFunctionImplementation)
+            functions.getBuiltin(ident.name(), ident.argumentTypes());
+        return tableFunction.execute(function.arguments().stream().map(a -> (Input) a).collect(Collectors.toList()));
+    }
+
+    public void assertExecute(String expr, String expected) {
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/tablefunctions/AbstractTableFunctionsTest.java]: same here?

File: sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        }
+
+        @Override
+        public FunctionInfo info() {
+            return info;
+        }
+
+        @Override
+        public Bucket execute(Collection<? extends Input> arguments) {
+            return new Bucket() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java]: why not `return CollectionBucket(Collections.singletonList(new Object[0]))` ?

File: dex/src/main/java/io/crate/data/Buckets.java [hunk 5]
```diff
@@ -52,9 +52,11 @@ public Row apply(Object[] input) {
 
     public static Object[][] materialize(Bucket bucket) {
         Object[][] res = new Object[bucket.size()][];
-        int i = 0;
-        for (Row row : bucket) {
-            res[i++] = row.materialize();
+        if (bucket.size() > 0) {
```
Comment (mfussenegger) [inline dex/src/main/java/io/crate/data/Buckets.java]: I'm not convinced that this is a perf improvement?

File: sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java (hunk 4 above)
Comment (andreidan) [inline sql/src/main/java/io/crate/operation/tablefunctions/BlankTableFunction.java]: Because I had no idea it existed. thanks

**Pull request merged**